solr/solrj/src/java/org/apache/solr/client/solrj/io/graph/ShortestPathStream.java

```diff
@@ -433,7 +433,7 @@ public JoinRunner(List<String> nodes) {
       StringBuffer nodeQuery = new StringBuffer();
 
       for(String node : nodes) {
-        nodeQuery.append(node).append(" ");
+        nodeQuery.append('"').append(node).append('"').append(" ");
```

Comment:
What happens if there are quotes in the id?

Reply:
Good remark

Reply:
> What happens if there are quotes in the id?

Good question! I'd briefly wondered about the backwards compatibility aspects w.r.t. adding surrounding quotes in the code when the caller already added them as a workaround, and then I had lost track of that.

One difference between the code adding surrounding quotes vs. the caller adding them is that the code also adds them for interim nodes. I've amended the test coverage to explore that further.

Also adjusted the code change to not add surrounding quotes if there are some already but obviously quotes could also be in the middle of the id, say `Alice ("Ally")` or `Robert ("Bob")` both of which also contain parentheses as it happens, hmm.